packages/react-instantsearch-hooks-dom/src/ui/Pagination.tsx

```diff
@@ -5,6 +5,49 @@ import { isModifierClick } from './lib/isModifierClick';
 
 import type { CreateURL } from 'instantsearch.js';
 
+export type PaginationTranslations = {
+  /**
+   * The label for the first page's button.
+   */
+  first: string;
+  /**
+   * The label for the previous page's button.
+   */
+  previous: string;
+  /**
+   * The label for the next page's button.
+   */
+  next: string;
+  /**
+   * The label for the last page's button.
+   */
+  last: string;
+  /**
+   * The label for a page's button.
+   */
+  page(currentPage: number): string;
```

Comment:
At this point why don't we accept components?

```suggestion
  page(props: { currentPage: number }): JSX.Element | string;
```

Reply:
Initially that wasn't supposed to be in the scope of the Translations API, that was the split between translations and templates (which for example we pass as single components in `<SearchBox>`.

As you suggested during our last call on the topic, we may want to reconsider the problem space of translations when it comes to the widget API. For now since it's internal, I don't think we need it, and we can keep the discussion for when we do the new Translations API RFC. wdyt?